pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java

```diff
@@ -1166,6 +1166,14 @@ public void updatePublishDispatcher() {
             }
         } else {
             log.info("Disabling publish throttling for {}", this.topic);
+            if (topicPublishRateLimiter != null
+                && topicPublishRateLimiter != PublishRateLimiter.DISABLED_RATE_LIMITER) {
```

Comment:
I think it's no need to test `topicPublishRateLimiter != PublishRateLimiter.DISABLED_RATE_LIMITER`

Reply:
OK, PTAL again @Technoboy-

Reply:
I have override the `close` method by https://github.com/apache/pulsar/pull/15529
So you need to rebase and no need to try/catch.